daemon/daemon.go

```diff
@@ -541,7 +544,8 @@ func servePrometheusMetrics(configParams *config.Config) {
 			}
 		}
 		http.Handle("/metrics", promhttp.Handler())
-		err := http.ListenAndServe(fmt.Sprintf(":%v", configParams.PrometheusMetricsPort), nil)
+		err := http.ListenAndServe(fmt.Sprintf("[%v]:%v",
```

Comment:
This is wrong for IPv4 addresses, isn't it - have you tested this?

Anyway, please use `net.JoinHostPort` instead.

Reply:
It works for all these: `localhost`, `127.0.0.1`, `::1`.

As for net.JoinHostPort, I'll look into that, thank you.

Reply:
I'm surprised that `localhost` and `127.0.0.1` work, but thanks for testing that it does.  Even so, I think I'd prefer net.JoinHostPort...

Reply:
I have modified the code accordingly and created PR's to do the same for HealthHost. See also:
https://github.com/projectcalico/typha/pull/285
https://github.com/projectcalico/libcalico-go/pull/1109/